pixy/consumer.go

```diff
@@ -0,0 +1,543 @@
+package pixy
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/mailgun/kafka-pixy/Godeps/_workspace/src/github.com/mailgun/kazoo-go"
+	"github.com/mailgun/kafka-pixy/Godeps/_workspace/src/github.com/mailgun/log"
+	"github.com/mailgun/kafka-pixy/Godeps/_workspace/src/github.com/mailgun/sarama"
+)
+
+type ErrConsumerSetup error
+type ErrConsumerBufferOverflow error
+type ErrConsumerRequestTimeout error
```

Comment:
If you want, you can group these into one declaration:

```go
type (
        ErrConsumerSetup error
        ErrConsumerBufferOverflow error
        ErrConsumerRequestTimeout error
)
```

Not a big deal either way, but I thought I would mention it in case you didn't know about grouped type declarations :)